src/operator/tensor/elemwise_binary_scalar_op_basic.cc

```diff
@@ -21,66 +21,76 @@
  * \file elemwise_binary_scalar_op.cc
  * \brief CPU Implementation of unary function.
  */
-#include "./elemwise_unary_op.h"
 #include "./elemwise_binary_op.h"
 #include "./elemwise_binary_scalar_op.h"
 
 namespace mxnet {
 namespace op {
 MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_plus_scalar)
-.set_attr<FCompute>("FCompute<cpu>", BinaryScalarCompute<cpu, mshadow::op::plus>)
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::BinaryScalarCompute<cpu, mshadow::op::plus>)
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_copy"})
 .add_alias("_PlusScalar");
 
 MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_minus_scalar)
-.set_attr<FCompute>("FCompute<cpu>", BinaryScalarCompute<cpu, mshadow::op::minus>)
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::BinaryScalarCompute<cpu, mshadow::op::minus>)
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_copy"})
 .add_alias("_MinusScalar");
 
 MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_rminus_scalar)
-.set_attr<FCompute>("FCompute<cpu>", BinaryScalarCompute<cpu, mshadow_op::rminus>)
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::BinaryScalarCompute<cpu, mshadow_op::rminus>)
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"negative"})
 .add_alias("_RMinusScalar");
 
 MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_mul_scalar)
-.set_attr<FCompute>("FCompute<cpu>", BinaryScalarCompute<cpu, mshadow::op::mul>)
-.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_mul_scalar"})
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::BinaryScalarCompute<cpu, mshadow::op::mul>)
+.set_attr<FComputeEx>("FComputeEx<cpu>", BinaryScalarOp::BinaryScalarComputeEx<
+  cpu, mshadow::op::mul>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_backward_mul_scalar"})
 .add_alias("_MulScalar");
 
+MXNET_OPERATOR_REGISTER_BINARY_SCALAR_DR(_backward_mul_scalar)
```

Comment:
I know that we're assuming dense ograd for now, but it will not always be the case. For future extension, I don't think backward_mul_scalar will already return a dense result. If ograd is row_sparse, the input grad should be row_sparse, too. Correct?